### PR TITLE
 feat(ui): add autocomplete for git.repository and git.tag as git.branch

### LINKED ISSUE #3218: feat(ui): add autocomplete for git.repository and git.tag as git.branch

### PR DESCRIPTION
**Pull request by bnjjj**
1. Description
1. Related issues
close #3218 
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

File: engine/api/application.go [hunk 1]
```diff
@@ -433,6 +433,65 @@ func (api *API) getApplicationBranchHandler() service.Handler {
...
+func (api *API) getApplicationVCSInfosHandler() service.Handler {
+	return func(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+		vars := mux.Vars(r)
+		projectKey := vars["key"]
+		applicationName := vars["permApplicationName"]
+		remote := r.FormValue("remote")
+
+		proj, err := project.Load(api.mustDB(), api.Cache, projectKey, getUser(ctx))
+		if err != nil {
+			return sdk.WrapError(err, "getApplicationBranchHandler> Cannot load project %s from db", projectKey, projectKey)
```
Comment (yesnault) [inline engine/api/application.go]: projectKey is write twice, probably missing application name

File: engine/vcs/github/client_tag.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+			nextPage = getNextPage(headers)
+		} else {
+			break
+		}
+	}
+
+	//Put the body on cache for one hour and one minute
+	g.Cache.SetWithTTL(cache.Key("vcs", "github", "tags", g.OAuthToken, "/repos/"+fullname+"/tags"), tags, 61*60)
+
+	fmt.Printf("tags %+v\n", tags)
```
Comment (yesnault) [inline engine/vcs/github/client_tag.go]: Printf is necessary :)

**Comment by ovh-cds** [comment]
CDS Report it#5660.0 ✘  
* w-it
  * w-it ✘

**Comment by ovh-cds** [comment]
CDS Report it#5665.0 ✘  
* w-it
  * w-it ✘

**Comment by ovh-cds** [comment]
CDS Report ut-engine#5667.0 ✘  
* Stage 1
  * Engine Test with DB ✘  
Unit Tests Report
* github.com/ovh/cds/cli


* github.com/ovh/cds/engine/api



















































































































































































  * Test_postWorkflowJobArtifactHandler ✘

* github.com/ovh/cds/engine/api/application








* github.com/ovh/cds/engine/api/environment




* github.com/ovh/cds/engine/api/keys



* github.com/ovh/cds/engine/api/migrate


* github.com/ovh/cds/engine/api/pipeline






* github.com/ovh/cds/engine/api/plugin


* github.com/ovh/cds/engine/api/project









* github.com/ovh/cds/engine/api/secret






* github.com/ovh/cds/engine/api/user








* github.com/ovh/cds/engine/api/warning















* github.com/ovh/cds/engine/api/worker










* github.com/ovh/cds/engine/api/workflow














































* github.com/ovh/cds/engine/hatchery/swarm



















* github.com/ovh/cds/engine/hooks






* github.com/ovh/cds/engine/repositories


* github.com/ovh/cds/engine/vcs











* github.com/ovh/cds/engine/vcs/bitbucket


















* github.com/ovh/cds/engine/vcs/github













* github.com/ovh/cds/engine/vcs/gitlab









* github.com/ovh/cds/engine/worker






* github.com/ovh/cds/sdk















* github.com/ovh/cds/sdk/exportentities




























* github.com/ovh/cds/sdk/interpolate

























* github.com/ovh/cds/sdk/luascript





* github.com/ovh/cds/sdk/plugin/dummy/caller


* github.com/ovh/cds/sdk/vcs/git

**Pull request merged**